### PR TITLE
Add validation to Patient Form

### LINKED ISSUE #709: Add error handling to Patient info form
Currently, if you do something that should trigger an error, like remove an entry from a required field, nothing happens.

### PR DESCRIPTION
**Pull request by ClydeDroid**
## Related Issue or Background Info
Resolves #709
Partially implements #380 

## Changes Proposed

- Adds validation on blur, fix, and submit to `PatientForm.tsx`

## Screenshots / Demos

https://user-images.githubusercontent.com/9121162/108910773-2db42380-75db-11eb-9168-e511edac023e.mp4

### CODE REVIEW

**Comment by maya** [comment]
I can't tell from the demo, but are you adding focus to the error messages on submit like we are on the time of test forms?

**Comment by ClydeDroid** [comment]
@maya I haven't added that yet, no. I'll add it to my list

**Comment by maya** [comment]
@ClydeDroid sounds good, this also led me to discover USWDS doesn't have a form summary component to show multiple errors. I opened an [issue in the USWDS repo](https://github.com/uswds/uswds/issues/4019) and planning to open an issue in our repo too.

File: frontend/src/app/patients/PatientForm.tsx [hunk 1]
```diff
@@ -248,16 +252,105 @@ const PatientForm = (props: Props) => {
...
       value = null;
     }
+    if (errors[name]) {
+      validateField(e);
+    }
     setFormChanged(true);
     setPatient({ ...patient, [e.target.name]: value });
   };
+
+  const validateField = (e: { target: any }, setState = true) => {
```
Comment (nullflux) [inline frontend/src/app/patients/PatientForm.tsx:262]: This implementation _works_ but is a little complicated – could you add some docs?

**Review by nullflux — APPROVED** [review]
This implementation is very tightly coupled with the form and its business logic – but it certainly meets the goals of the ticket! Given what a sprint it is to get this thing in front of users, I'm ✅  with making validation "elegant" as tech debt later on

**Pull request merged**